torchvision/prototype/transforms/_misc.py

```diff
@@ -225,28 +226,99 @@ def _transform(
+    # - small or degenerate bboxes based on min_size (this includes those where X2 <= X1 or Y2 <= Y1)
+    # - boxes with any coordinate outside the range of the image (negative, or > spatial_size)
+    _transformed_types = datapoints.BoundingBox
 
-    def __init__(self, min_size: float = 1.0) -> None:
+    def __init__(self, min_size: float = 1.0, labels="default") -> None:
         super().__init__()
+
+        if min_size < 1:
+            raise ValueError(f"min_size must be >= 1, got {min_size}.")
         self.min_size = min_size
 
-    def _get_params(self, flat_inputs: List[Any]) -> Dict[str, Any]:
-        bounding_box = query_bounding_box(flat_inputs)
-
-        # TODO: We can improve performance here by not using the `remove_small_boxes` function. It requires the box to
-        #  be in XYXY format only to calculate the width and height internally. Thus, if the box is in XYWH or CXCYWH
-        #  format,we need to convert first just to afterwards compute the width and height again, although they were
-        #  there in the first place for these formats.
-        bounding_box = F.convert_format_bounding_box(
-            bounding_box.as_subclass(torch.Tensor),
-            old_format=bounding_box.format,
+        self.labels = labels
+        if labels == "default":
+            self._get_labels = self._find_label_default_heuristic
+        elif callable(self.labels):
+            self._get_labels = labels
+        elif isinstance(self.labels, str):
+            self._get_labels = lambda inputs: inputs[labels]
+        else:
+            raise ValueError(
+                "labels parameter should either be a str, callable, or 'default'. "
+                f"Got {labels} of type {type(labels)}."
+            )
+
+    def _find_label_default_heuristic(self, inputs):
+        # Tries to find a "label" key, otherwise tries for the first key that contains "label" - case insensitive
+        # Returns None if nothing is found
+        labels = None
+        candidate_key = None
+        with suppress(StopIteration):
+            candidate_key = next(key for key in inputs.keys() if key.lower() == "labels")
+        if candidate_key is None:
+            with suppress(StopIteration):
+                candidate_key = next(key for key in inputs.keys() if "label" in key.lower())
+        labels = inputs.get(candidate_key)
+
+        if labels is None:
+            raise ValueError(
+                "Could not infer where the labels are in the sample. Try passing a callable as the label parameter?"
+            )
+        return labels
+
+    def forward(self, *inputs: Any) -> Any:
+        inputs = inputs if len(inputs) > 1 else inputs[0]
+
+        if isinstance(self.labels, str) and not isinstance(inputs, dict):
+            raise ValueError(
+                f"If labels is a str or 'default' (got {self.labels}), then the input to forward() must be a dict. "
+                f"Got {type(inputs)} instead"
+            )
```

Comment:
Maybe I don't understand what you have in mind, but we have to do this check here (in fowrad()) anyway because it must have the `dict()`. To this end, I'd rather having it explicitly here than "monkey-patch" that check into both `_find_label_default_heuristic` and the `lambda`?

Reply:
Unless I miss something, this check is only here for the case of user supplied string keys, right? Can't we replace the lambda in the constructor with

```python
def _find_label_str_heuristic(inputs):
    if not isinstance(inputs, dict):
        raise ValueError(
            f"If labels is a str or 'default' (got {self.labels}), then the input to forward() must be a dict. "
            f"Got {type(inputs)} instead"
        )

     return inputs[self.labels]
```

Reply:
> this check is only here for the case of user supplied string keys

The check is valid for `"default"` as well